modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -249,6 +189,46 @@ public Exception getInboundException() {
         return inboundException;
     }
 
+    protected io.netty.handler.codec.http.HttpRequest getNettyRequest() {
+        return request;
+    }
+
+    protected static RestRequest.Method translateRequestMethod(HttpMethod httpMethod) {
+        if (httpMethod == HttpMethod.GET) return RestRequest.Method.GET;
+
+        if (httpMethod == HttpMethod.POST) return RestRequest.Method.POST;
+
+        if (httpMethod == HttpMethod.PUT) return RestRequest.Method.PUT;
+
+        if (httpMethod == HttpMethod.DELETE) return RestRequest.Method.DELETE;
+
+        if (httpMethod == HttpMethod.HEAD) {
+            return RestRequest.Method.HEAD;
+        }
+
+        if (httpMethod == HttpMethod.OPTIONS) {
+            return RestRequest.Method.OPTIONS;
+        }
+
+        if (httpMethod == HttpMethod.PATCH) {
+            return RestRequest.Method.PATCH;
+        }
+
+        if (httpMethod == HttpMethod.TRACE) {
+            return RestRequest.Method.TRACE;
+        }
+
+        if (httpMethod == HttpMethod.CONNECT) {
+            return RestRequest.Method.CONNECT;
+        }
+
+        throw new IllegalArgumentException("Unexpected http method: " + httpMethod);
+    }
+
+    protected static Map<String, List<String>> wrapHttpHeaders(HttpHeaders httpHeaders) {
+        return new HttpHeadersMap(httpHeaders);
+    }
```

Comment:
I need to reuse `translateRequestMethod` and `wrapHttpHeaders` when building the `BasicHttpRequest` in the new channel handler.
This way the `HttpRequest` instances, which are built for the complete request (including the body contents), and the `BasicHttpRequest` instances, which are built for the body-less requests from the new channel handler, reuse the same code.